packages/modal/src/Modal.js

```diff
@@ -0,0 +1,126 @@
+import React, { useEffect } from "react";
+import ReactDOM from "react-dom";
+import PropTypes from "prop-types";
+import { useTransition, animated } from "react-spring";
+import Link from "@crave/farmblocks-link";
+
+import {
+  Wrapper,
+  Overlay,
+  ConstrainedCard,
+  ContentWrapper,
+  Header,
+} from "./Modal.styled";
+
+const Modal = ({
+  isOpen,
+  parentNode,
+  shouldCloseOnOverlayClick,
+  shouldCloseOnEsc,
+  onRequestClose,
```

Comment:
What about `onClose`?

Reply:
I think it would be misleading. Because the component is stateless, the events that trigger this prop don't actually close the modal. We have to manage the `isOpen` prop to make it close.

`onClose` could be called when the modal did close, after `isOpen` change to `false`. https://github.com/CraveFood/farmblocks/pull/800#discussion_r288758286

Reply:
Hmm yeah, it makes sense. Actually we would never use this `onRequestClose`, right? Only if we didn't use the helpers to render the modal.

We are good here... Thanks for the clarification.